Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.h

```diff
@@ -83,6 +83,30 @@ WK_CLASS_AVAILABLE(macos(10.11), ios(9.0))
 /*! @abstract Returns the cookie store representing HTTP cookies in this website data store. */
 @property (nonatomic, readonly) WKHTTPCookieStore *httpCookieStore WK_API_AVAILABLE(macos(10.13), ios(11.0));
 
+/*! @abstract Get identifier for a data store.
+ @discussion Default and non-persistent data store do not have identifier.
+ */
+@property (nonatomic, readonly, nullable) NSUUID *identifier WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
+
+/*! @abstract Get a persistent data store.
+ @param identifier An identifier that is used to uniquely identify the data store.
+ @discussion If data store with specified identifier does not exists yet, it will be created. It returns nil if data store cannot be created.
```

Comment:
If data store => If a data store
exists yet => exist yet
It returns => Returns

Can we specify the conditions that would prevent a data store from being created?

Reply:
Yes, if identifier is invalid.